tabular/src/autogluon/tabular/models/tabular_nn/categorical_encoders.py

```diff
@@ -166,7 +166,7 @@ def _check_X(self, X):
         """
         if not (hasattr(X, 'iloc') and getattr(X, 'ndim', 0) == 2):
             # if not a dataframe, do normal check_array validation
-            X_temp = check_array(X, dtype=None)
+            X_temp = check_array(X, dtype=None, force_all_finite=False)
             if (not hasattr(X, 'dtype')
                     and np.issubdtype(X_temp.dtype, np.str_)):
                 X = check_array(X, dtype=np.object)
```

Comment:
Is the same `force_all_finite` fix required here as well?

Reply:
Theoretically I think so but I don't know in what situation that line is ever accessed, and I'd rather the failure be demonstrated in a benchmark before trying to fix it.